frontend/src/features/permits/hooks/hooks.ts

```diff
@@ -54,38 +54,41 @@ export const useSaveTermOversizeMutation = () => {
 /**
  * A custom react query hook that get application details from the backend API
  * The hook gets application data by its permit ID
+ * @param applicationStep The step of the application process (form, review, or pay)
  * @param permitId permit id for the application, if it exists
+ * @param permitType permit type for the application, if it exists
  * @returns appropriate Application data, or error if failed
  */
 export const useApplicationDetailsQuery = (
   applicationStep: ApplicationStep,
   permitId?: string,
+  permitType?: string,
 ) => {
   const [applicationData, setApplicationData] =
     useState<Nullable<Application>>();
 
-  // Currently, creating new application route doesn't contain permitId
-  // ie. route === "/applications/new" instead of "/applications/:permitId"
-  // Thus we need to do a check
-  // Also, "/applications/new" is valid, but "/applications/new/review" is invalid
-  // Thus we also need applicationStep to determine which page (route) we're on
+  // Check if the permit type param is valid, if there is one
+  const permitTypeValid = isPermitTypeValid(permitType);
+
   const isCreateNewApplication =
-    typeof permitId === "undefined" &&
+    permitTypeValid &&
     applicationStep === APPLICATION_STEPS.DETAILS;
 
-  // Check if the permit id is numeric
-  const isPermitIdNumeric = () => {
-    if (!permitId) return false;
-    if (permitId.trim() === "") return false;
-    return !isNaN(Number(permitId.trim()));
-  };
-
-  const isInvalidRoute = !isPermitIdNumeric() && !isCreateNewApplication;
-  const shouldEnableQuery = isPermitIdNumeric();
+  // Currently, creating new application route doesn't contain valid permitId
```

Comment:
How can new application route contain a valid permitId? Isn't it created upon first save?

Reply:
Ok so both the create new application ("/create-application/:permitType") and application details for after creation/save ("/applications/:permitId") shares the same hook and ApplicationForm component, except one has the "permitType" param in the url (create new) and the other has permitId (after saved, which is checked by isPermitIdNumeric)

Reply:
The create new application indeed doesn't have valid permitId (undefined from useParams), but it does have valid permitType which is used by isCreateNewApplication in this instance, and also the shouldEnableQuery would be false which means that it won't fetch application details from the backend (as expected) since it's a new application and there's nothing to fetch